File: WeekTwo/DayTwo/vehicle.js
class Vechicle {
    constructor(colour, make, year, lights) {
        this.colour = colour
        this.make = make
        this.year = year
        this.lights = lights
    }

    honk_horn() {
        if (this.lights == true) {
            console.log("Your lights are on so I am honking my horn")
        }
        else {
            console.log("Your lights are off there is no need for me to honk")
        }
    }
}

const car = new Vechicle("red", "honda", 2014, true)
car.honk_horn()

class Truck extends Vechicle {
    constructor(colour, make, year, lights, name) {
        super(colour, make, year, lights)
        this.name = name
    }

    popWheelie() {
        console.log("This is dangerous.")
    }
}

const truck = new Truck("green", "Volvo", 2020, true, "Betsy")
console.log(truck.name)

File: WeekFour/DayOne/Classes/Vehicle.cs
namespace Practice
{
    public class Vehicle
    {
        public string make;
        public string colour;
        public int year;

        public Vehicle(string Make, string Colour, int Year)
        {
            make = Make;
            colour = Colour;
            year = Year;
        }

        public void display()
        {
            Console.WriteLine($"My car make is {make}. My car colour is {colour}. I am honking my horn.");
        }
    }

    // public class Truck : Vehicle
    // {
    //     // public Truck() : base(Make, Colour, Year)
    //     // {
    //     //     make = Make;
    //     //     colour = color
    //     // }
    // }
}